google/cloud/storage/bucket_metadata_test.cc

```diff
@@ -338,6 +387,31 @@ TEST(BucketMetadataTest, ResetEncryption) {
   EXPECT_THAT(os.str(), ::testing::Not(::testing::HasSubstr("encryption.")));
 }
 
+/// @test Verify we can reset the Object Lifecycle in BucketMetadata.
+TEST(BucketMetadataTest, ResetLifecycle) {
+  auto expected = CreateBucketMetadataForTest();
+  auto copy = expected;
+  EXPECT_TRUE(copy.has_lifecycle());
+  copy.reset_lifecycle();
+  EXPECT_FALSE(copy.has_lifecycle());
+  EXPECT_NE(expected, copy);
+  std::ostringstream os;
+  os << copy;
+  EXPECT_THAT(os.str(), ::testing::Not(::testing::HasSubstr("lifecycle.")));
```

Comment:
Nit: I think the `::testing::` bit here is distracting.  It also conflicts in style with others tests that use `HasSubstr` by itself. This would read more smoothly as `Not(HasSubstr(...))`.

On that note, you could just add using-declarations at the top of the anonymous namespace so that all the tests have access to those testing symbols.

Reply:
Fixed.